### PR TITLE
Research DNA disintegrating content

### PR DESCRIPTION
**Pull request by iloveloopers**
# About the pull request
monkeys can now catch the xeno transforming disease given by DDI
now only need 150 cloneloss instead of 190 to trigger the disease
now turns you into a mutated drone (freed mob if monkey) that cannot evolve to queen
marines no longer get 10 tech points for scanning DDI

if a xeno is made through the disease while inside research containment, marines get rewarded with 1 tech point per minute for a span of 20 minutes

# Explain why it's good for the game
cool feature and adds a reason to hide your DDI vial from the goons
actually lets you test DDI without being bwoinked for LRP self testing
more research experimenting and new and exciting content

# Testing Photographs and Procedure
yes I tested this and (I think) all edge cases

# Changelog
:cl:
add: Monkeys can now catch the xeno transforming disease from DDI and will give a reward if they transform while inside research containment.
add: Individuals infected with xeno transformation disease will now play xeno sounds intermittently.
balance: DDI no longer gives 10 tech points when scanned.
add: DDI now transforms you into a mutated drone instead of a random selection of normal hive tier 1s
/:cl:

### CODE REVIEW

File: code/datums/diseases/xeno_transformation.dm [hunk 1]
```diff
@@ -36,18 +37,36 @@
...
+				else
+					var/mob/living/carbon/human/H = affected_mob
+					H.Alienize(XENO_T3_CASTES)
 				src.cure()
 
+				if(istype(area, /area/almayer/medical/containment))
+					if(!GLOB.chemical_data.DDI_experiment_done)
+						GLOB.chemical_data.DDI_experiment_done = TRUE
+
+						ai_announcement("Notice: XX-121 entity created at research containment through unknown means, analyzing data...")
```
Comment (Diegoflores31) [inline code/datums/diseases/xeno_transformation.dm:66]: this breaks lore . ARES is unable to detect Xenos

File: code/datums/diseases/xeno_transformation.dm [hunk 2]
```diff
@@ -36,18 +37,36 @@
...
 			if(prob(40))
-				var/turf/T = find_loc(affected_mob)
-				gibs(T)
-				var/mob/living/carbon/human/H = affected_mob
-				H.Alienize(XENO_T1_CASTES)
+				var/turf/turf = get_turf(affected_mob)
+				var/area/area = get_area(turf)
+				gibs(turf)
+				if(ismonkey(affected_mob))
+					new /mob/living/simple_animal/hostile/alien/ravager(turf)
```
Comment (Diegoflores31) [inline code/datums/diseases/xeno_transformation.dm:55]: i would recomend you not doing this ? pretty sure mob xenos are not supposed to be in normal gameplay

File: code/datums/diseases/xeno_transformation.dm (hunk 1 above)
Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm]: fixed

File: code/datums/diseases/xeno_transformation.dm (hunk 2 above)
Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm:55]: they work fine though? their AI works and they target and slash humans

**Comment by RainbowStalin** [comment]
I can see a future where goons just become an arrest on sight for most COs/XOs. Since now theres an actual insentive to fuck them up besided "I hate WY".

File: code/datums/diseases/xeno_transformation.dm (hunk 2 above)
Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm]: turned into feral xeno

**Comment by iloveloopers** [comment]
should be ready now

**Comment by Danil154322** [comment]
One more way of mutilating each other for our beloved researchers! Let's goooo!!!

**Comment by Skatuk** [comment]
10 credits is there for the purchase of the chemical, now goons have no reason to take it since it'd basically be a robbery.

File: code/datums/diseases/xeno_transformation.dm [hunk 3]
```diff
@@ -36,18 +37,32 @@
...
-				var/mob/living/carbon/human/H = affected_mob
-				H.Alienize(XENO_T1_CASTES)
-				src.cure()
+				var/turf/turf = get_turf(affected_mob)
+				var/area/area = get_area(turf)
+				gibs(turf)
+				for(var/obj/item/items in affected_mob)
+					affected_mob.drop_inv_item_on_ground(items)
+				var/mob/living/carbon/xenomorph/drone/xeno = new()
+				xeno.forceMove(turf)
```
Comment (kiVts) [inline code/datums/diseases/xeno_transformation.dm:57]:
```suggestion
				var/mob/living/carbon/xenomorph/drone/xeno = new(turf)
```

File: code/datums/diseases/xeno_transformation.dm [hunk 4]
```diff
@@ -36,18 +37,32 @@
...
+				xeno.set_hive_and_update(XENO_HIVE_FERAL)
+				if(ismonkey(affected_mob))
+					xeno.free_for_ghosts(TRUE)
+				else
+					affected_mob.mind.transfer_to(xeno, TRUE)
```
Comment (kiVts) [inline code/datums/diseases/xeno_transformation.dm:62]:
```suggestion
					affected_mob.mind?.transfer_to(xeno, TRUE)
```

File: code/modules/reagents/chemical_research/research-events.dm [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+	ai_announcement("Notice: Unidentified lifesign detected at research containment created through DNA disintegration, analyzing data...")
+	sleep(5 SECONDS)
+	ai_announcement("Notice: Lifeform biostructural data can be analyzed further for tech point reward at a rate of 1 point per minute.")
+
+	timer = world.time
+	START_PROCESSING(SSprocessing, src)
+
+/datum/research_event/ddi_experiment/process(delta_time)
+	if(total_points_given >= 20)
```
Comment (kiVts) [inline code/modules/reagents/chemical_research/research-events.dm:31]: you REALLY dont want to hardcode this, define is our lord and savior

File: code/datums/diseases/xeno_transformation.dm (hunk 4 above)
Comment (kiVts) [reply code/datums/diseases/xeno_transformation.dm:62]: may be actually unneeded but doesnt seem like so

Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm]: not sure this really matters but sure

File: code/modules/reagents/chemical_research/research-events.dm [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+/datum/research_event/ddi_experiment/process(delta_time)
+	if(total_points_given >= MAX_POINTS)
+		STOP_PROCESSING(SSprocessing, src)
+		ai_announcement("Notice: Lifeform biostructural data fully analyzed, 20 total tech points awarded. Recommend termination of lifeform.")
+		return
+
+	if(QDELETED(linked_xeno)) //they must have evolved
+		if(QDELETED(player.mob)) //no new mob? they got deleted
+			ai_announcement("Notice: Lifeform terminated or missing, biostructural data not fully analyzed. Only [total_points_given] out of [20] tech points awarded.")
```
Comment (kiVts) [inline code/modules/reagents/chemical_research/research-events.dm:39]:
```suggestion
			ai_announcement("Notice: Lifeform terminated or missing, biostructural data not fully analyzed. Only [total_points_given] out of [MAX_POINTS] tech points awarded.")
```

File: code/modules/reagents/chemical_research/research-events.dm [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+			STOP_PROCESSING(SSprocessing, src)
+			return
+		linked_xeno = player.mob //get the updated mob from the client
+
+	var/area/xeno_loc = get_area(linked_xeno.loc)
+
+	if(linked_xeno.stat == DEAD || !istype(xeno_loc, /area/almayer/medical/containment)) //you let it escape or die. idiot
+		if(total_points_given < MAX_POINTS)
+			ai_announcement("Notice: Lifeform terminated or missing, biostructural data not fully analyzed. Only [total_points_given] out of [20] tech points awarded.")
```
Comment (kiVts) [inline code/modules/reagents/chemical_research/research-events.dm:48]:
```suggestion
			ai_announcement("Notice: Lifeform terminated or missing, biostructural data not fully analyzed. Only [total_points_given] out of [MAX_POINTS] tech points awarded.")
```

Comment (kiVts) [reply code/modules/reagents/chemical_research/research-events.dm:48]: why was 20 []'ed anyway

File: code/modules/reagents/chemical_research/research-events.dm [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+/datum/research_event/ddi_experiment/proc/trigger(xeno)
+	if(DDI_experiment_triggered)
+		return
+	DDI_experiment_triggered = TRUE
+	linked_xeno = xeno
+	if(linked_xeno.client)
+		player = linked_xeno.client
+	RegisterSignal(linked_xeno, COMSIG_MOB_NEW_MIND, PROC_REF(get_mind)) //this works as long as the client is not forcefully set by admins
+
+	ai_announcement("Notice: Unidentified lifesign detected at research containment created through DNA disintegration, analyzing data...")
```
Comment (Diegoflores31) [inline code/modules/reagents/chemical_research/research-events.dm:24]: DNA dising is a top secret stuff its impossible that ares (and old USCM AI) knows about it.
Change it to something like ...
"Unknown lifesign detected in contaiment .... Analizing"

File: code/modules/reagents/chemical_research/research-events.dm [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+	sleep(5 SECONDS)
+	ai_announcement("Notice: Lifeform biostructural data can be analyzed further for tech point reward at a rate of 1 point per minute.")
+
+	timer = world.time
+	START_PROCESSING(SSprocessing, src)
+
+/datum/research_event/ddi_experiment/process(delta_time)
+	if(total_points_given >= MAX_POINTS)
+		STOP_PROCESSING(SSprocessing, src)
+		ai_announcement("Notice: Lifeform biostructural data fully analyzed, 20 total tech points awarded. Recommend termination of lifeform.")
```
Comment (Diegoflores31) [inline code/modules/reagents/chemical_research/research-events.dm:34]: Dont force the termination . Let it be

File: code/datums/diseases/xeno_transformation.dm [hunk 10]
```diff
@@ -36,18 +37,31 @@
...
-				var/mob/living/carbon/human/H = affected_mob
-				H.Alienize(XENO_T1_CASTES)
-				src.cure()
+				var/turf/turf = get_turf(affected_mob)
+				var/area/area = get_area(turf)
+				gibs(turf)
+				for(var/obj/item/items in affected_mob)
+					affected_mob.drop_inv_item_on_ground(items)
+				var/mob/living/carbon/xenomorph/drone/xeno = new(turf)
+				xeno.set_hive_and_update(XENO_HIVE_FERAL)
```
[the remark this answers is not in the document]
Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm]: done but I'm not certain Drathek will approve since mutated hive can have a queen

[the remark this answers is not in the document]
Comment (Diegoflores31) [reply code/datums/diseases/xeno_transformation.dm]: Does it matter ? You can already grow queen with way less effort and current DDI can grow queens too

[the remark this answers is not in the document]
Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm]: alright, it now checks if the mutated hive already had members and forbids evolving to queen if it doesn't

File: code/datums/diseases/xeno_transformation.dm [hunk 11]
```diff
@@ -55,6 +55,8 @@
 					affected_mob.drop_inv_item_on_ground(items)
 				var/mob/living/carbon/xenomorph/drone/xeno = new(turf)
 				xeno.set_hive_and_update(XENO_HIVE_MUTATED)
+				if(length(xeno.hive.totalXenos) <= 1)
```
Comment (Diegoflores31) [inline code/datums/diseases/xeno_transformation.dm:58]: Isnt this disabling queen evolve for the entire hive ??

Comment (iloveloopers) [reply code/datums/diseases/xeno_transformation.dm:58]: Yes, if the hive has no members other than the DDI xeno. Not really a big deal, won't affect events where admins make the base hive mutated. Could possibly affect thunderdome I guess?

**Comment by iloveloopers** [comment]
this says a lot about CM
